lib/plugins/aws/deployFunction/index.js

```diff
@@ -164,6 +164,10 @@ class AwsDeployFunction {
             const errorMessage = 'Invalid characters in environment variable';
             throw new this.serverless.classes.Error(errorMessage);
           }
+          if (typeof params.Environment.Variables[key] !== 'string') {
```

Comment:
Does that also still work for CF references like this?
```
Environment:
  MY_VAR:
    Ref: myResource
```
It is important that CF references that are resolved while CloudFormation deploys the stack still work.
Maybe it would be better to error only on disallowed integral types with a condition like this:
`if (_.isNumber(...) || _.isBoolean(...) || ... ` or even a specific check on the special AWS CF functions.
E.g. check for anything that is not a string and not an object that contains `Ref`, or `Fn::*`:

```
if (!_.isString(...) || _.isObject(...) && !_.some(..., (val, key) => _.startsWith(key, 'Fn::'))) {
``` 

Additionally you should use lodash here for all typechecks.